84.md

```diff
@@ -0,0 +1,98 @@
+NIP-84
+======
+
+Highlights
+----------
+
+`draft` `optional` `author:pablof7z`
+
+This NIP defines `kind:9802`, a "highlight" event, to signal content a user finds valuable.
+
+## Format
+The `.content` of these events is the highlighted portion of the text.
+
+`.content` might be empty for highlights of non-text based media (e.g. NIP-94 audio/video).
+
+### References
+Events SHOULD tag the source of the highlight, whether nostr-native or not.
+`a` or `e` tags should be used for nostr events and `r` tags for URLs.
+
+When tagging a URL, clients generating these events SHOULD do a best effort of cleaning the URL from trackers
+or obvious non-useful information from the query string.
+
+### Attribution
+Clients MAY include one or more `p` tags, tagging the original authors of the material being highlighted; this is particularly
+useful when highlighting non-nostr content for which the client might be able to get a nostr pubkey somehow
+(e.g. prompting the user or reading a `<meta name="nostr:pubkey" />` tag on the document). A role MAY be included as the
+last value of the tag.
+
+```json
+[ "p", "...", "author" ],
+[ "p", "...", "author" ],
+[ "p", "...", "editor" ],
+```
+
+### Context
+Clients MAY include a `context` tag, useful when the highlight is a subset of a paragraph and displaying the
+surrounding content might be beneficial to give context to the higlight.
+
+### Ranges
+Clients MAY include `range` tags with the start/end indexes of where the highlight begins and finishes within
+the referenced article/tagged-event.
+
+```
+[ "range", <start-index>, <end-index> ]
+```
+
+Additionally a range with `context` as the third value of the tag MAY be added to indicate the begin/finish indexes
+of the highlight within the included `context` tag.
+
+```
+[ "range", <start-index>, <end-index>, "context" ]
+```
+
+#### Text-based nostr events' highlights
+
+Highlights of nostr-natives events SHOULD use the range index of the content as-is
+(e.g. NIP-23 articles include the markdown instead of computing the index from the rendered markdown).
+
+```
+[ "range", 3000, 3042 ] # highlight begins at index position 3000 of the tagged event's `.content`
+[ "range", 42, 84, "context" ] # highlight begins at index position 42 of the `context` tag's value
+```
+
+#### Non-text-based nostr events's highlights
```

Comment:
events'

Reply:
damn it! 😂 thanks

Reply:
You missed a chance to use the most amazing feature of the English language.